test/e2e_node/hugepages_test.go

```diff
@@ -121,24 +116,39 @@ func configureHugePages() error {
 		return err
 	}
 	e2elog.Logf("HugePages_Total is set to %v", numHugePages)
-	if numHugePages == 50 {
+	if int64(numHugePages) == pageCount {
 		return nil
 	}
-	return fmt.Errorf("expected hugepages %v, but found %v", 50, numHugePages)
+	return fmt.Errorf("expected hugepages %v, but found %v", pageCount, numHugePages)
 }
 
 // releaseHugePages releases all pre-allocated hugepages
 func releaseHugePages() error {
 	return exec.Command("/bin/sh", "-c", "echo 0 > /proc/sys/vm/nr_hugepages").Run()
 }
 
-// isHugePageSupported returns true if the default hugepagesize on host is 2Mi (i.e. 2048 kB)
-func isHugePageSupported() bool {
+// getDefaultHugePageSize returns the default huge page size, and a boolean if huge pages are supported
+func getDefaultHugePageSize() (resource.Quantity, bool) {
 	outData, err := exec.Command("/bin/sh", "-c", "cat /proc/meminfo | grep 'Hugepagesize:' | awk '{print $2}'").Output()
 	framework.ExpectNoError(err)
 	pageSize, err := strconv.Atoi(strings.TrimSpace(string(outData)))
 	framework.ExpectNoError(err)
-	return pageSize == 2048
+	if pageSize == 0 {
+		return resource.Quantity{}, false
+	}
+	return *resource.NewQuantity(int64(pageSize*1024), resource.BinarySI), true
+}
+
+func getTestValues() (hugePageSize resource.Quantity, totalMemory resource.Quantity, pageCount int64) {
+	hugePageSize, _ = getDefaultHugePageSize()
```

Comment:
changing the test to support a broader set of hugepage configurations is a nice update.  thanks for this.  2MB and 1Gi were most common when this feature was authored on platforms that had desired it.

Reply:
Thanks! I guess 99% will use the default 2MB size, but you will never know...

Another thing outside the scope of the PR: https://github.com/kubernetes/kubernetes/pull/50859/files#diff-f19a7777a36882029db668a427f950feR3517 Is there actually a reason to throw this error? It makes it "impossible" to start kubelet propery when having more than one huge page size with allocated memory. I know that adding support for using multiple page sizes for a pod would require quite a bit work, but each node should be able to support multiple sizes, or am I wrong? Cant find anything about it in the KIP either.. https://github.com/kubernetes/enhancements/tree/master/keps/sig-node

Reply:
supporting more than one page size per node was requested by @dchen1107 when we first introduced this feature.  in general, i am not aware of how common it is for a workload to demand more than one page size, and nodes that require pre-allocated hugepages are probably purpose built.